aws/credentials/credentials.go

```diff
@@ -255,3 +271,16 @@ func (c *Credentials) IsExpired() bool {
 func (c *Credentials) isExpired() bool {
 	return c.forceRefresh || c.provider.IsExpired()
 }
+
+// ExpiresAt provides access to the functionality of the Expirer interface of
+// the underlying Provider, if it supports that interface.  Otherwise, it returns
+// an error.
+func (c *Credentials) ExpiresAt() (time.Time, error) {
+	expirer, ok := c.provider.(Expirer)
+	if !ok {
+		return time.Time{}, awserr.New("ProviderNotExpirer",
+			fmt.Sprintf("provider %s does not support ExpiresAt()", reflect.TypeOf(c.provider)),
+			nil)
+	}
+	return expirer.ExpiresAt(), nil
```

Comment:
I think accessing the `expirer.ExpiersAt` needs to be protected with the mutex similar to `Credentials .IsExpired` on like 163. Otherwise, there could be a race updating the expires time and reading it.

Reply:
In addition I think the `Credentials`'s `forceRefresh` flag should be considered here. If the credentials have been flagged as as needing to be refreshed then I think they should also be considered as already expired.

Maybe call `isExpired()` first and returning `time.Now()` if the credentials are flagged as expired. Otherwise return the actual expiry time.

Reply:
It wasn't clear to me that forceRefresh (i.e. Credentials.Expire()) is actually used outside of test programs or example programs.  But, can check forceRefresh, and return a zero time.Time{} in that case.